File: android/src/main/java/com/evotor/egais/mapper/Ticket.kt
package com.evotor.egais.mapper

import android.database.Cursor
import com.evotor.egais.converter.MarkListConverter
import com.evotor.egais.converter.QuantityBigDecimalConverter
import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.WritableMap

internal fun createTicket(cursor: Cursor): WritableMap {
    val result = Arguments.createMap()
    result.putString("uuid", cursor.getString(cursor.getColumnIndexOrThrow("UUID")))
    result.putString("owner", cursor.getString(cursor.getColumnIndexOrThrow("OWNER")))
    result.putString("ticketDate", cursor.getString(cursor.getColumnIndexOrThrow("TICKET_DATE")))
    result.putString("identity", cursor.getString(cursor.getColumnIndexOrThrow("IDENTITY")))
    result.putString("docId", cursor.getString(cursor.getColumnIndexOrThrow("DOC_ID")))
    result.putString("transportId", cursor.getString(cursor.getColumnIndexOrThrow("TRANSPORT_ID")))
    result.putString("regId", cursor.getString(cursor.getColumnIndexOrThrow("REG_ID")))
    result.putString("docHash", cursor.getString(cursor.getColumnIndexOrThrow("DOC_HASH")))
    result.putString("docType", cursor.getString(cursor.getColumnIndexOrThrow("DOC_TYPE")))
    result.putString("conclusionType", cursor.getString(cursor.getColumnIndexOrThrow("RESULT_CONCLUSION")))
    result.putString("resultDate", cursor.getString(cursor.getColumnIndexOrThrow("RESULT_DATE")))
    result.putString("resultComments", cursor.getString(cursor.getColumnIndexOrThrow("RESULT_COMMENTS")))
    result.putString("operationResultName", cursor.getString(cursor.getColumnIndexOrThrow("OPERATION_RESULT_NAME")))
    result.putString("operationResultResult", cursor.getString(cursor.getColumnIndexOrThrow("OPERATION_RESULT_RESULT")))
    result.putString("operationResultComment", cursor.getString(cursor.getColumnIndexOrThrow("OPERATION_RESULT_COMMENT")))
    result.putString("operationResultDate", cursor.getString(cursor.getColumnIndexOrThrow("OPERATION_RESULT_DATE")))
    return result
}

internal fun createConfirmTicket(cursor: Cursor): WritableMap {
    val result = Arguments.createMap()
    result.putString("uuid", cursor.getString(cursor.getColumnIndexOrThrow("UUID")))
    result.putString("owner", cursor.getString(cursor.getColumnIndexOrThrow("OWNER")))
    result.putString("identity", cursor.getString(cursor.getColumnIndexOrThrow("IDENTITY")))
    result.putString("isConfirm", cursor.getString(cursor.getColumnIndexOrThrow("IS_CONFIRM")))
    result.putString("ticketNumber", cursor.getString(cursor.getColumnIndexOrThrow("TICKET_NUMBER")))
    result.putString("ticketDate", cursor.getString(cursor.getColumnIndexOrThrow("TICKET_DATE")))
    result.putString("wbRegId", cursor.getString(cursor.getColumnIndexOrThrow("WB_REG_ID")))
    result.putString("note", cursor.getString(cursor.getColumnIndexOrThrow("NOTE")))
    result.putString("status", cursor.getString(cursor.getColumnIndexOrThrow("STATUS")))
    result.putString("rejectComment", cursor.getString(cursor.getColumnIndexOrThrow("REJECT_COMMENT")))
    return result
}

File: model/document/Ticket.js
/**
 * Классы и типы для работы с квитанциями.
 * @module module:ticket
 */
import {ConclusionType, ConfirmTicketStatus, DocType, OperationResult} from '../types'

/**
 * @class module:ticket.Ticket
 * @classdesc Квитанция.
 * @param {string} uuid Уникальный идентификатор
 * @param {string} docOwner Кто подает документы
 * @param {Date} ticketDate Дата квитанции
 * @param {?string} identity Идентификатор документа(исходного, клиентский из содержимого XML(может не быть))
 * @param {?string} docId Ид задачи из файловой базы
 * @param {?string} transportId Транспортный ид из файловой базы
 * @param {?string} regId Присвоенный документу номер
 * @param {?string} docHash Хэш документа
 * @param {?module:ticket#DocType} docType Тип исходного документа
 * @param {?module:ticket#ConclusionType} resultConclusion Результат фиксации
 * @param {?Date} resultDate Дата фиксации
 * @param {?string} resultComments Описание результата фиксации, описание ошибки фиксации
 * @param {?string} operationResultName Операция
 * @param {?module:ticket#OperationResult} operationResultResult Результат выполнения
 * @param {?string} operationResultComment Сообщение
 * @param {?Date} operationResultDate Дата операции
 */
export default class Ticket {
    constructor(uuid: string,
                docOwner: string,
                ticketDate: (Date | null),
                identity: (string | null),
                docId: (string | null),
                transportId: (string | null),
                regId: (string | null),
                docHash: (string | null),
                docType: (DocType | null),
                resultConclusion: (ConclusionType | null),
                resultDate: (Date | null),
                resultComments: (string | null),
                operationResultName: (string | null),
                operationResultResult: (OperationResult | null),
                operationResultComment: (string | null),
                operationResultDate: (Date | null)) {
        this.uuid = uuid;
        this.docOwner = docOwner;
        this.ticketDate = ticketDate;
        this.identity = identity;
        this.docId = docId;
        this.transportId = transportId;
        this.regId = regId;
        this.docHash = docHash;
        this.docType = docType;
        this.resultConclusion = resultConclusion;
        this.resultDate = resultDate;
        this.resultComments = resultComments;
        this.operationResultName = operationResultName;
        this.operationResultResult = operationResultResult;
        this.operationResultComment = operationResultComment;
        this.operationResultDate = operationResultDate;
    }
}

/**
 * @class module:ticket.ConfirmTicket
 * @classdesc Подтверждение акта разногласий для Товарно-Транспортной Накладной.
 * @param {string} uuid Уникальный идентификатор
 * @param {string} docOwner Кто подает документы
 * @param {?string} identity Идентификатор документа (клиентский, к заполнению необязательный)
 * @param {?module:ticket#ConclusionType} isConfirm Тип подтверждения: Принимаем/отказываем
 * @param {?string} ticketNumber Номер подтверждения
 * @param {?Date} ticketDate Дата составления подтверждения
 * @param {?string} wbRegId ИД накладной в системе
 * @param {?string} note примечание
 * @param {module:ticket#ConfirmTicketStatus} status Статус
 * @param {?string} rejectComment Комментарий для отказа
 */
export class ConfirmTicket {
    constructor(uuid: string,
                docOwner: string,
                identity: (string | null),
                isConfirm: (ConclusionType | null),
                ticketNumber: (string | null),
                ticketDate: (Date | null),
                wbRegId: (string | null),
                note: (string | null),
                status: ConfirmTicketStatus,
                rejectComment: (string | null)) {
        this.uuid = uuid;
        this.docOwner = docOwner;
        this.identity = identity;
        this.isConfirm = isConfirm;
        this.ticketNumber = ticketNumber;
        this.ticketDate = ticketDate;
        this.wbRegId = wbRegId;
        this.note = note;
        this.status = status;
        this.rejectComment = rejectComment;
    }
}